nucypher-core-wasm/tests/wasm.rs

```diff
@@ -663,3 +663,87 @@ fn metadata_response() {
         "MetadataResponse does not roundtrip"
     );
 }
+
+//
+// ThresholdDecryptionRequestResponse
+//
+
+#[wasm_bindgen_test]
+fn threshold_decryption_request() {
+    let ritual_id: u16 = 5;
+    let request_secret = SecretKey::random();
+    let request_encrypting_key = request_secret.public_key();
+
+    let response_secret = SecretKey::random();
+    let response_encrypting_key = response_secret.public_key();
+
+    let conditions: JsValue = Some(Conditions::new("{'some': 'condition'}")).into();
+    let context: JsValue = Some(Context::new("{'user': 'context'}")).into();
+
+    let request = ThresholdDecryptionRequest::new(
+        ritual_id,
+        0,
+        b"fake ciphertext",
+        &conditions.unchecked_into::<OptionConditions>(),
+        &context.unchecked_into::<OptionContext>(),
+    )
+    .unwrap();
+
+    let encrypted_request = request.encrypt(&request_encrypting_key, &response_encrypting_key);
+
+    let encrypted_request_bytes = encrypted_request.to_bytes();
+    let encrypted_request_from_bytes =
+        EncryptedThresholdDecryptionRequest::from_bytes(&encrypted_request_bytes).unwrap();
+
+    assert_eq!(encrypted_request_from_bytes.id(), ritual_id);
+
+    let e2e_request = encrypted_request_from_bytes
+        .decrypt(&request_secret)
+        .unwrap();
+    assert_eq!(
+        response_encrypting_key.to_compressed_bytes(),
+        e2e_request.response_encrypting_key().to_compressed_bytes()
+    );
+    assert_eq!(request, e2e_request.decryption_request());
+
+    // wrong secret key used
+    assert!(encrypted_request_from_bytes
+        .decrypt(&response_secret)
```

Comment:
Just a quick question: Mistaking `response_secret` for a `request_secret` wouldn't be possible IRL, right? Because there is one for each party involved in communication, and the requestor wouldn't have access to the responder?